### PR TITLE
[Improve][CI] improve ci checking

### PR DESCRIPTION
**Pull request by SbloodyS**
## Purpose of the pull request

1. add ```docs``` path exclude from ```unit-test``` and ```e2e-test``` and ```backend```
2. add ```dolphinscheduler-ui``` and ```dolphinscheduler-ui-next``` path exclude from ```backend```
3. add ```docs``` path include from ```docs-test```
4. add ```find ./docs``` in ```docs-test```

### CODE REVIEW

File: .github/workflows/e2e.yml [hunk 1]
```diff
@@ -17,9 +17,13 @@
 
 on:
   pull_request:
+    paths-ignore:
+      - 'docs/**'
   push:
     branches:
       - dev
+    paths-ignore:
+      - 'docs/**'
```
Comment (kezhenxu94) [inline .github/workflows/e2e.yml]: I remember I told you that this can not be added because if the workflow is skipped, the PRs cannot be merged

**Review by kezhenxu94 — CHANGES_REQUESTED** [review]

File: .github/workflows/e2e.yml (hunk 1 above)
Comment (SbloodyS) [reply .github/workflows/e2e.yml]: Sorry,I forgot this. I'll see if i can fix it. If not i will remove it. @kezhenxu94

Comment (SbloodyS) [reply .github/workflows/e2e.yml]: I try to modify ```e2e-test```  with ```https://github.com/dorny/paths-filter#examples``` and ```https://github.com/dorny/paths-filter/issues/97```. Please check if this is correct. If not i will remove it. Thanks. @kezhenxu94

Comment (SbloodyS) [reply .github/workflows/e2e.yml]: It seems like ```dorny/paths-filter@v2 is not allowed to be used in apache/dolphinscheduler```. Sad...

Comment (SbloodyS) [reply .github/workflows/e2e.yml]: I have submitted a request to apache infra to see if this plugin can be enable. ```https://issues.apache.org/jira/browse/INFRA-23069```

In the mean time, I will mark this pr as draft until this plugin is enable or i can find another way to solve this problem.

Comment (kezhenxu94) [reply .github/workflows/e2e.yml]: @SbloodyS you can use this getsentry/paths-filter@v2 instead, seems they have the same functionalities and this is approved already by infra team as I know

Comment (SbloodyS) [reply .github/workflows/e2e.yml]: > @SbloodyS you can use this getsentry/paths-filter@v2 instead, seems they have the same functionalities and this is approved already by infra team as I know

Wow, It seems working. I will do more tests to verify it. Thanks.

**Comment by SbloodyS** [comment]
@kezhenxu94 @zhongjiajie Please review. Thanks.

**Review by kezhenxu94 — APPROVED** [review]

File: .github/workflows/backend.yml [hunk 2]
```diff
@@ -63,3 +77,17 @@ jobs:
                  -Dmaven.wagon.httpconnectionManager.ttlSeconds=120
       - name: Check dependency license
         run: tools/dependencies/check-LICENSE.sh
+  result:
+    name: Build
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    needs: [ build ]
```
Comment (zhongjiajie) [inline .github/workflows/backend.yml:84]: Did we miss job `paths-filter` dependent here?

Comment (SbloodyS) [reply .github/workflows/backend.yml:84]: It does not need to. It already exists in steps.run.

Comment (kezhenxu94) [reply .github/workflows/backend.yml:84]: @zhongjiajie we don't need explicit dependent here, `result` depends on `build`, `build` depends on `paths-filter`, so `result` depends on `paths-filter` implicitly

Comment (zhongjiajie) [reply .github/workflows/backend.yml:84]: Thanks for the clarification.

**Pull request merged**